app/views/unattended/provisioning_templates/finish/windows_default_finish.erb

```diff
@@ -52,15 +57,22 @@ params:
   w32tm /resync
   w32tm /resync
 
-  :: You can join your machine to the domain right here >
-
-  <% if host_param('domainAdminAccount') and host_param('domainAdminAccountPasswd') -%>
-    echo joining domain
-    powershell.exe -OutputFormat text -ExecutionPolicy remotesigned -command c:\deploy\joinDomain.ps1
+  <% if host_param('computerDomain') -%>
+    <% if host_param('domainAdminAccount').present? and host_param('domainAdminAccountPasswd').present? -%>
+      echo performing secure domain join
+      echo $Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList ([pscustomobject]@{ UserName = '<%= host_param('domainAdminAccount') -%>'; Password = (ConvertTo-SecureString -String '<%= host_param('domainAdminAccountPasswd') -%>' -AsPlainText -Force)[0] }) > c:\deploy\joinDomain.ps1
```

Comment:
Does c:\deploy exist at this point? I am just curious.

Reply:
yes, the script that calls the powershell is located there too